app/models/user.rb

```diff
@@ -30,6 +30,16 @@ class User < ApplicationRecord
   has_many :groups,
     through: :group_owners
 
+  # NOTE(ezekg) This association is only used to preload a user's status, since
+  #             the #status needs to check if a user has any active licenses.
+  #
+  #             We're doing a DISTINCT ON so that we can query ANY one active
+  #             license for each user, since users can potentially have thousands
+  #             and thousands of licenses, and superfluously preloading/querying
+  #             that many records would a bad idea.
+  has_one :any_active_license, -> { active.reorder(nil).distinct_on(:user_id) },
```

Comment:
I love Postgres. This would've been so complicated with MySQL using a `LATERAL JOIN`.